client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -1280,9 +1309,11 @@ def find_string_differences(files: List[str]) -> Dict[str, str]:
     # Find common prefix using zip_longest to compare all characters at once
     prefix = ""
     for chars in zip_longest(*processed_files):
-        if len(set(chars) - {None}) != 1:  # If there's more than one unique character
+        # If there's more than one unique character
```

Comment:
I'm really confused about this part.

I think it does not do what you want to do. It could skip characters if `None` is first in the set. And it really needs more readabilty.

I think this was the intention?
```python
for chars in zip_longest(*processed_files):
    chars_s = set(chars)
    # Ignore shorter base names
    chars_s.discard(None)
    # End if a character in filenames is not the same
    if len(chars_s) != 1:
        break
    prefix += next(iter(chars_s))
```

Suffix lookup does the same.